### PR TITLE
Fix typehints in library controllers

### PR DESCRIPTION
**Pull request by wilsonge**
### Summary of Changes
Now we have void typehints with our PHP 7.2 requirement we can add void typehints into the main library Controllers

### Testing Instructions
Code review

### Documentation Changes Required
None (these typehints are already in the code and will be documented appropriately)

### CODE REVIEW

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 911f2274a0c77551593fffc9f4eea7e4a3d84e63<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28477">issues.joomla.org/tracker/joomla-cms/28477</a>.</sub>

File: libraries/src/MVC/Controller/ApiController.php [hunk 1]
```diff
@@ -86,13 +86,13 @@ class ApiController extends BaseController
...
 	 * @param   MvcFactoryInterface  $factory  The factory.
-	 * @param   CmsApplication       $app      The JApplication for the dispatcher
+	 * @param   CMSApplication       $app      The JApplication for the dispatcher
 	 * @param   Input                $input    Input
 	 *
 	 * @since   4.0.0
 	 * @throws  \Exception
 	 */
-	public function __construct($config = array(), MvcFactoryInterface $factory = null, $app = null, $input = null)
+	public function __construct($config = array(), MvcFactoryInterface $factory = null, ?CMSApplication $app = null, ?Input $input = null)
```
Comment (SharkyKZ) [inline libraries/src/MVC/Controller/ApiController.php]: Can be done in a separate PR but `MvcFactoryInterface` should be capitalized as `MVCFactoryInterface`.

Comment (wilsonge) [reply libraries/src/MVC/Controller/ApiController.php]: Fixed :)

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on 3b0eb70f258696e25f1799479720a5394a3048af<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28477">issues.joomla.org/tracker/joomla-cms/28477</a>.</sub>
1st and 2nd one

**Comment by wilsonge** [comment]
Thanks guys!

**Pull request merged**